python/cvmfs/repository.py

```diff
@@ -344,156 +509,19 @@ def _retrieve_and_open_catalog(self, catalog_hash):
-
-
-
-class RemoteRepository(Repository):
-    """ Concrete Repository implementation for a repository reachable by HTTP """
-    def __init__(self, repo_url):
-        self._storage_location = urlparse.urlunparse(urlparse.urlparse(repo_url))
-        self._try_to_get_repo_information()
-        Repository.__init__(self)
-
-
-    def __str__(self):
-        return self._storage_location
-
-    def __repr__(self):
-        return "<Remote Repository " + self.fqrn + " at " + self._storage_location + ">"
-
-
-    def _get_rest_url(self, method_name):
-        return "/".join((self._storage_location,
-                         _common._REST_CONNECTOR,
-                         method_name))
-
-
-    def has_rest_api(self):
-        if not hasattr(self, '_rest_api'):
-            api_url = self._get_rest_url('info')
-            response = requests.head(api_url)
-            self._has_rest_api = (response.status_code == requests.codes.ok)
-        return self._has_rest_api
-
-
-    def __rest_request(self, http_verb_method, rest_method_name):
-        api_url  = self._get_rest_url(rest_method_name)
-        response = http_verb_method(api_url)
-        response.raise_for_status()
-        return response.json()
-
-
-    def _GET_rest_request(self, method_name):
-        return self.__rest_request(requests.get, method_name)
-
-
-    def _POST_rest_request(self, method_name):
-        return self.__rest_request(requests.post, method_name)
-
-
-    def _try_to_get_repo_information(self):
-        self.type          = 'unknown'
-        self.version       = 'unknown'
-        if self.has_rest_api():
-            general_infos      = self._GET_rest_request('info')
-            self.type          = general_infos['type']
-            self.version       = general_infos['version']
-
-
-    def start_replication(self):
-        res = self._POST_rest_request('replicate')
-        if res['result'] != 'ok':
-            raise CannotReplicate(self)
-
-
-    def retrieve_file(self, file_name):
-        file_url = self._storage_location + "/" + file_name
-        tmp_file = tempfile.NamedTemporaryFile('w+b')
-        response = requests.get(file_url, stream=True)
-        if response.status_code != requests.codes.ok:
-            raise FileNotFoundInRepository(self, file_url)
-        for chunk in response.iter_content(chunk_size=4096):
-            if chunk:
-                tmp_file.write(chunk)
-        tmp_file.seek(0)
-        tmp_file.flush()
-        return tmp_file
-
-
-
 def all_local():
     d = _common._REPO_CONFIG_PATH
     if not os.path.isdir(d):
         raise _common.CvmfsNotInstalled
-    return [ LocalRepository(repo) for repo in os.listdir(d) if os.path.isdir(os.path.join(d, repo)) ]
+    return [ Repository(cache_dir=repo) for repo in os.listdir(d) if os.path.isdir(os.path.join(d, repo)) ]
 
 def all_local_stratum0():
     return [ repo for repo in all_local() if repo.type == 'stratum0' ]
 
 def open_repository(repository_path, public_key = None):
```

Comment:
This was just a convenience function to distinguish between the old `LocalRepository` and `RemoteRepository` classes, based on the provided URL/path. This should be handled by the generic `Repository` class now (see above). Please remove it entirely.